cmd/commands/ibc.go

```diff
@@ -21,172 +20,110 @@ func NewIBCPlugin() *ibc.IBCPlugin {
-		Name:  "chain_id",
-		Usage: "ChainID for the new blockchain",
-		Value: "",
-	}
-
-	IbcGenesisFlag = cli.StringFlag{
-		Name:  "genesis",
-		Usage: "Genesis file for the new blockchain",
-		Value: "",
-	}
-
-	IbcHeaderFlag = cli.StringFlag{
-		Name:  "header",
-		Usage: "Block header for an ibc update",
-		Value: "",
-	}
-
-	IbcCommitFlag = cli.StringFlag{
-		Name:  "commit",
-		Usage: "Block commit for an ibc update",
-		Value: "",
-	}
-
-	IbcFromFlag = cli.StringFlag{
-		Name:  "from",
-		Usage: "Source ChainID",
-		Value: "",
+	IBCTxCmd = &cobra.Command{
+		Use:   "ibc",
+		Short: "An IBC transaction, for InterBlockchain Communication",
 	}
 
-	IbcToFlag = cli.StringFlag{
-		Name:  "to",
-		Usage: "Destination ChainID",
-		Value: "",
+	IBCRegisterTxCmd = &cobra.Command{
+		Use:   "register",
+		Short: "Register a blockchain via IBC",
+		RunE:  ibcRegisterTxCmd,
 	}
 
-	IbcTypeFlag = cli.StringFlag{
-		Name:  "type",
-		Usage: "IBC packet type (eg. coin)",
-		Value: "",
+	IBCUpdateTxCmd = &cobra.Command{
+		Use:   "update",
+		Short: "Update the latest state of a blockchain via IBC",
+		RunE:  ibcUpdateTxCmd,
 	}
 
-	IbcPayloadFlag = cli.StringFlag{
-		Name:  "payload",
-		Usage: "IBC packet payload",
-		Value: "",
+	IBCPacketTxCmd = &cobra.Command{
+		Use:   "packet",
+		Short: "Send a new packet via IBC",
 	}
 
-	IbcPacketFlag = cli.StringFlag{
-		Name:  "packet",
-		Usage: "hex-encoded IBC packet",
-		Value: "",
+	IBCPacketCreateTxCmd = &cobra.Command{
+		Use:   "create",
+		Short: "Create an egress IBC packet",
+		RunE:  ibcPacketCreateTxCmd,
 	}
 
-	IbcProofFlag = cli.StringFlag{
-		Name:  "proof",
-		Usage: "hex-encoded proof of IBC packet from source chain",
-		Value: "",
-	}
-
-	IbcSequenceFlag = cli.IntFlag{
-		Name:  "sequence",
-		Usage: "sequence number for IBC packet",
-		Value: 0,
+	IBCPacketPostTxCmd = &cobra.Command{
+		Use:   "post",
+		Short: "Deliver an IBC packet to another chain",
+		RunE:  ibcPacketPostTxCmd,
 	}
+)
 
-	IbcHeightFlag = cli.IntFlag{
-		Name:  "height",
-		Usage: "Height the packet became egress in source chain",
-		Value: 0,
-	}
+//flags
+var (
+	ibcChainIDFlag  string
+	ibcGenesisFlag  string
+	ibcHeaderFlag   string
+	ibcCommitFlag   string
+	ibcFromFlag     string
+	ibcToFlag       string
+	ibcTypeFlag     string
+	ibcPayloadFlag  string
+	ibcPacketFlag   string
+	ibcProofFlag    string
+	ibcSequenceFlag int
+	ibcHeightFlag   int
 )
 
-//---------------------------------------------------------------------
-// ibc commands
+func init() {
 
-var (
-	IbcTxCmd = cli.Command{
-		Name:  "ibc",
-		Usage: "an IBC transaction, for InterBlockchain Communication",
-		Flags: TxFlags,
-		Subcommands: []cli.Command{
-			IbcRegisterTxCmd,
-			IbcUpdateTxCmd,
-			IbcPacketTxCmd,
-		},
+	// register flags
+	registerFlags := []Flag2Register{
+		{&ibcChainIDFlag, "ibc_chain_id", "", "ChainID for the new blockchain"},
+		{&ibcGenesisFlag, "genesis", "", "Genesis file for the new blockchain"},
```

Comment:
actually leave this for now and we can discuss as a team after some QA testing. fine to have this on develop before a final decision